### PR TITLE
Improve locking safety for Registry

### PR DESCRIPTION
**Pull request by kvark**
This isn't ideal: we shouldn't be mutably locking anything for adding/removing objects in `local` path. But at least it makes it safe, for now.

### CODE REVIEW

**Comment by kvark** [comment]
To clarify, even though parts of this PR are to be scrapped, at least the change of `lock() -> read() + write()` will likely to stay.

File: wgpu-bindings/Cargo.toml [hunk 1]
```diff
@@ -10,4 +10,5 @@ authors = [
 default = []
 
 [dependencies]
-cbindgen = { git = "https://github.com/eqrion/cbindgen.git" }
+#cbindgen = "0.6.7"
+cbindgen = { git = "https://github.com/eqrion/cbindgen" }
```
Comment (grovesNL) [inline wgpu-bindings/Cargo.toml]: We could also use [my branch](https://github.com/eqrion/cbindgen/pull/248) for now if we're blocked by the header changes

File: wgpu-native/src/resource.rs [hunk 2]
```diff
@@ -7,7 +7,6 @@ use hal;
 
 
 bitflags! {
-    #[repr(transparent)]
```
Comment (grovesNL) [inline wgpu-native/src/resource.rs]: I assume this is temporary?

File: wgpu-native/src/registry/remote.rs [hunk 3]
```diff
@@ -1,19 +1,16 @@
...
-use parking_lot::{Mutex, MutexGuard};
-use std::sync::Arc;
 
 
 pub type Id = u32;
-pub type ItemsGuard<'a, T> = MutexGuard<'a, Items<T>>;
 
 pub struct Items<T> {
-    next_id: Id,
+    next_id: u32,
```
Comment (grovesNL) [inline wgpu-native/src/registry/remote.rs]: Why `Id` -> `u32` here?

File: wgpu-native/src/registry/remote.rs [hunk 4]
```diff
@@ -35,11 +32,11 @@ impl<T> super::Items<T> for Items<T> {
         id
     }
 
-    fn get(&self, id: Id) -> super::Item<T> {
+    fn get(&self, id: Id) -> & T {
```
Comment (grovesNL) [inline wgpu-native/src/registry/remote.rs]: nit: space

File: wgpu-native/src/registry/remote.rs (hunk 3 above)
Comment (kvark) [reply wgpu-native/src/registry/remote.rs]: refactor artifact :)

File: wgpu-native/src/resource.rs (hunk 2 above)
Comment (kvark) [reply wgpu-native/src/resource.rs]: accidental...

**Comment by kvark** [comment]
Thanks for the review! All the concerns should be addressed now.
bors r=grovesNL

**Pull request merged**